tests/core/scenario/test_scenario_manager.py

```diff
@@ -326,14 +327,44 @@ def test_notification_subscribe(mocker):
     # test unsubscribing notification
     # test notis subscribe only on new jobs
     # _ScenarioManager._get(scenario)
-    _ScenarioManager._unsubscribe(notify_1, scenario)
+    _ScenarioManager._unsubscribe(callback=notify_1, scenario=scenario)
     _ScenarioManager._subscribe(callback=notify_2, scenario=scenario)
     _ScenarioManager._submit(scenario)
 
     notify_1.assert_not_called()
     notify_2.assert_called_3_times()
 
 
+def test_notification_subscribe_multiple_params():
+    Config.configure_job_executions(mode=JobConfig._DEVELOPMENT_MODE)
+    _Scheduler._update_job_config()
+
+    scenario_config = Config.configure_scenario(
+        "awesome_scenario",
+        [
+            Config.configure_pipeline(
+                "by_6",
+                [
+                    Config.configure_task(
+                        "mult_by_2",
+                        mult_by_2,
+                        [Config.configure_data_node("foo", "in_memory", Scope.PIPELINE, default_data=1)],
+                        Config.configure_data_node("bar", "in_memory", Scope.SCENARIO, default_data=0),
+                    )
+                ],
+            )
+        ],
+    )
+
+    scenario = _ScenarioManager._create(scenario_config)
+    _ScenarioManager._subscribe(callback=notify_multi_param, params=["foobar", 123, 1.2], scenario=scenario)
+    _ScenarioManager._submit(scenario)
+
```

Comment:
I would find a way to assert that the submit method actually calls the callback.